libraries/script-engine/src/ScriptEngine.cpp

```diff
@@ -1119,8 +1235,14 @@ void ScriptEngine::timerFired() {
 
     // call the associated JS function, if it exists
     if (timerData.function.isValid()) {
+        auto thisObject = timerData.function;
+        auto entityID = timerData.definingEntityIdentifier;
+        if (!entityID.isNull() && _entityScripts.contains(entityID)) {
+            EntityScriptDetails details = _entityScripts[entityID];
+            thisObject = details.scriptObject;
+        }
         auto preTimer = p_high_resolution_clock::now();
-        callWithEnvironment(timerData.definingEntityIdentifier, timerData.definingSandboxURL, timerData.function, timerData.function, QScriptValueList());
+        callWithEnvironment(entityID, timerData.definingSandboxURL, timerData.function, thisObject, QScriptValueList());
```

Comment:
This change is somewhat unrelated, but involves the same code areas -- it updates `Script.setTimeout`/`Script.setInterval` to use a default "this" context of the Entity instance for the callback (instead of using the callback function itself as "this").